packages/apptive_grid_core/lib/src/model/entity/entities_response.dart

```diff
@@ -24,6 +46,68 @@ class EntitiesResponse<T> {
   EntitiesResponse copyWith({List<T>? items}) {
     return EntitiesResponse(
       items: items ?? this.items,
+      pageMetaData: pageMetaData,
+    );
+  }
+}
+
+/// A class representing a response containing the paging meta data.
+class PageMetaData {
+  /// Constructs a new [PageMetaData] instance from a JSON [Map].
+  factory PageMetaData.fromJson(dynamic json) {
+    return PageMetaData(
+      numberOfItems: json['numberOfItems'],
+      numberOfPages: json['numberOfPages'],
+      size: json['size'],
+      page: json['page'],
     );
   }
+
+  /// Constructs a new [PageMetaData].
+  ///
+  /// [numberOfItems] is the total number of entities in the response.
+  /// [numberOfPages] is the total number of pages in the response.
+  /// [size] is the number of entities in each page.
+  /// [page] is the current page index (starting at 1).
+  const PageMetaData({
+    required this.numberOfItems,
+    required this.numberOfPages,
+    required this.size,
+    required this.page,
+  }) : super();
+
+  /// The total number of entities available.
+  final int numberOfItems;
+
+  /// The total number of pages available.
+  final int numberOfPages;
+
+  /// The number of entities in each page.
```

Comment:
Is this for _each_ page or rather for _this_ page? I'd assume if I have a page with 5 items because it's the last page/first page with only that amount it would be lower?

Reply:
I checked with Cyrille, its for **_each_** page.